app/src/main/java/utils/QueryUtils.java

```diff
@@ -74,18 +73,43 @@ public void done(List<BeachGroup> groups, ParseException e) {
                 for (BeachGroup group : groups) {
                     Log.i(TAG, "Group: " + group.getKeyGroupName());
                 }
-                favorite_beaches.addAll(groups);
-                Log.i(TAG, favorite_beaches.toString());
+                SpinnerAdapter beachAdapter = new SpinnerAdapter(view.getContext(), R.layout.activity_custom_spinner, groups);
+                spinnerBeach.setAdapter(beachAdapter);
             }
         });
-        return favorite_beaches;
     }
 
-    public static List<BeachGroup> queryBeaches() {
-        List<BeachGroup> beaches = new ArrayList<>();
+    public static void queryBeachesforGroups(BeachGroup beach, Button favoriteButton, Button favoriteButtonPressed){
+        ParseQuery<FavoriteGroups> groupsQuery = ParseQuery.getQuery(FavoriteGroups.class)
+                .include(FavoriteGroups.KEY_USER)
+                .whereEqualTo(FavoriteGroups.KEY_USER, ParseUser.getCurrentUser());
+        ParseQuery<BeachGroup> beachQuery = ParseQuery.getQuery(BeachGroup.class)
+                .whereEqualTo(BeachGroup.KEY_GROUP, beach)
+                .whereMatchesKeyInQuery(BeachGroup.KEY_GROUP, FavoriteGroups.KEY_GROUP, groupsQuery);
+        beachQuery.findInBackground(new FindCallback<BeachGroup>() {
+            @Override
+            public void done(List<BeachGroup> groups, ParseException e) {
+                String queryBeachName;
```

Comment:
My rule of thumb is that variables should be declared as close as possible to their use. Helps to avoid this:
int foo;
...
1000 lines of code
...
foo = 2;

It helps with code reading since if you see a new variable it's a signal that it's important and about to be used.

This would mean moving 
String queryBeachName;
String currentBeachName = beach.getKeyGroupName();

down into the for loop:
 for (BeachGroup group : groups) {
                    String queryBeachName = group.getKeyGroupName();
                    Log.i(TAG, "Group: " + queryBeachName);
                    String currentBeachName = beach.getKeyGroupName();
                    if(queryBeachName.equals(currentBeachName)) {
                        favoriteButtonPressed.setVisibility(View.VISIBLE);
                        favoriteButton.setVisibility(View.GONE);
                    }
                }